.github/workflows/update-submodules.yml

```diff
@@ -0,0 +1,42 @@
+name: Update Library Submodules
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Target branch for raising PR'
+        required: true
+        default: main
+      ignored:
+        description: 'Regex pattern passed to `grep -v` to ignore in the path to the library, e.g. "ota\|jobs" ignores all libraries with "ota" or "jobs" in the name.'
+        required: false
+        default: 'ignore_none'
+
+jobs:
+  update-submodules:
+    name: Update submodules
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Configure Git identity
+        run: |
+          git config --global user.name "Submodule Updater"
+      - name: Update the submodules
+        run: |
+          libs=$(find libraries/standard libraries/aws -maxdepth 1 -mindepth 1 | grep -v ${{ github.event.inputs.ignored }})
+          git submodule update --remote $libs
+      - name: Commit changes and Push to remote
+        run: |
+          now="$(date +'%d-%m-%Y')"
+          branch="updater-job/submodules-$now"
+          git checkout -b $branch
+          git commit -am 'Updating library submodules to the latest.'
+          git push --set-upstream origin $branch
+      - name: Raise a Pull-Request
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh pr create --base ${{ github.event.inputs.branch }} --title 'Update library submodules to the latest' --body 'Update library submodules.'
```

Comment:
If only a specific list of libraries' submodules are advanced, it may be good to enlist them in the PR name and description to make the PR information specific

Reply:
I had got rid of that hoping no one would care. but since you care i will put it back.

Reply:
This is too hard to do. Nothing I tried work.